core/src/test/java/io/aiven/klaw/service/UsersTeamsControllerServiceTest.java

```diff
@@ -381,7 +458,114 @@ void addNewUserInvalidPatternFailure() throws KlawException {
   }
 
   @Test
-  void addNewTeam() {}
+  void addNewTeamSuccess() throws KlawException {
+    int tenantId = 101;
+    String userName = "testuser";
+    String existingTeamName = "existingTeamName";
+    TeamModel teamModel = utilMethods.getTeamModelMock();
+    ArgumentCaptor<Team> teamCaptor = ArgumentCaptor.forClass(Team.class);
+
+    when(mailService.getUserName(any())).thenReturn(userName);
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(tenantId);
+    when(manageDatabase.getTeamNamesForTenant(tenantId)).thenReturn(List.of(existingTeamName));
+    when(handleDbRequests.addNewTeam(teamCaptor.capture())).thenReturn(ApiResultStatus.SUCCESS.value);
+
+    ApiResponse apiResponse = usersTeamsControllerService.addNewTeam(teamModel, true);
+    Team capturedTeam = teamCaptor.getValue();
+
+    assertThat(capturedTeam.getRequestTopicsEnvs()).isEqualTo(
```

Comment:
You might be able to extract this into a separate method for validation, that can be re-used to reduce code repetition

Reply:
You can also extract out the captured team and teamModel as well to reduce code duplication, something like. 

private void validateTeams(Team actual, TeamModel expected, int expectedTenantId) {
    assertThat(actual.getRequestTopicsEnvs())
        .isEqualTo(String.join(",", expected.getEnvList().toArray(new String[0])));
    assertThat(actual.getTeamname()).isEqualTo(expected.getTeamname());
    assertThat(actual.getTeamphone()).isEqualTo(expected.getTeamphone());
    assertThat(actual.getContactperson()).isEqualTo(expected.getContactperson());
    assertThat(actual.getTeamId()).isEqualTo(expected.getTeamId());
    assertThat(actual.getTenantId()).isEqualTo(expectedTenantId);
  }

Reply:
I have tried extracting as much code as possible into common private methods including for test setups to reduce the overall duplication.